variation-commons-core/src/test/java/uk/ac/ebi/eva/commons/core/models/AbstractVariantTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.ac.ebi.eva.commons.core.models;
+
+import org.hamcrest.CoreMatchers;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+
+import static org.junit.Assert.assertEquals;
+
+public class AbstractVariantTest {
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Test
+    public void testNegativeStartCoordinateInConstructor() throws IllegalArgumentException {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage(CoreMatchers.equalTo("Variant start coordinate cannot be negative"));
+        new Variant("1", -1, 1, "C", "T");
+    }
+
+    @Test
+    public void testNegativeStartCoordinateInRenormalization() throws IllegalArgumentException {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage(CoreMatchers.equalTo("Variant start coordinate cannot be negative"));
+        Variant variant = new Variant("1", 1, 1, "C", "T");
+        variant.renormalize(-1, 1, "A", "T");
+    }
+
+    @Test
+    public void testNegativeEndCoordinateInConstructor() throws IllegalArgumentException {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage(CoreMatchers.equalTo("Variant end coordinate cannot be negative"));
+        new Variant("1", 1, -1, "C", "T");
+    }
+
+    @Test
+    public void testNegativeEndCoordinateInRenormalization() throws IllegalArgumentException {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage(CoreMatchers.equalTo("Variant end coordinate cannot be negative"));

```

Comment:
@jmmut should these checks be placed after the variant object has been constructed? I remember there was some weird situation with code being executed after the checks.

Reply:
checking after the constructor would be a bit more correct, I think. here we check even the exception message, so I think we're mostly safe. for instance, it could go wrong if we put `new Variant("1", -1, 1, "C", "T");` by accident and the constructor is the one throwing the exception

Reply:
The documentation says to use expect before the code that throws exception https://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html and in fact, when I shuffle the order, the test does not work.